dev/src/bulk-writer.ts

```diff
@@ -975,10 +978,24 @@ export class BulkWriter {
       this._successFn.bind(this)
     );
 
+    // Swallow the error if the developer has set an error listener. This
+    // prevents UnhandledPromiseRejections from being thrown if a floating
+    // BulkWriter operation promise fails when an error handler is specified.
+    //
+    // This is done here in order to chain the caught promise onto `lastOp`,
+    // which ensures that flush() resolves after the operation promise.
+    const returnedBulkWriterPromise = bulkWriterOp.promise.catch(err => {
```

Comment:
This should probably use "op" instead of BulkWriter since it only affects a single operation.  

I would have also used the term "user" in this variable, but that may be the lack of oxygen at my current elevation (`userPromise`).